app/views/RoomView/index.tsx

```diff
@@ -515,7 +515,7 @@ class RoomView extends React.Component<IRoomViewProps, IRoomViewState> {
 				if (data[0]) {
 					if (this.subObserveQuery && this.subObserveQuery.unsubscribe) {
 						this.observeRoom(data[0]);
-						this.setState({ room: data[0] });
+						this.setState({ room: data[0], joined: true });
```

Comment:
When the user is joining the room from the thread, the main room didn't update correctly, because the state param `roomUpdate` didn't update yet, because this will update just after a new change inside the function `observeRoom` https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/29a71f4a5492f6f427827474f4493a121cb55d03/app/views/RoomView/index.tsx#L771-L778
Also, the state `joined` didn't update too, because this state is updated in the function `onJoin` https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/29a71f4a5492f6f427827474f4493a121cb55d03/app/views/RoomView/index.tsx#L1075-L1079